src/tls/cockpit-certificate-helper.in

```diff
@@ -26,6 +26,65 @@ install_key() {
     fi
 }
 
+selfsign_sscg() {
+    sscg --quiet \
+        --lifetime "${DAYS}" \
+        --key-strength 2048 \
+        --cert-key-file "${KEYFILE}" \
+        --cert-file "${CERTFILE}" \
+        --ca-file "${CA_FILE}" \
+        --hostname "${HOSTNAME}" \
+        --organization "${MACHINE_ID}" \
+        --subject-alt-name localhost \
+        --subject-alt-name IP:127.0.0.1/255.255.255.255
+}
+
+selfsign_openssl() {
+    openssl req -x509 \
+        -days "${DAYS}" \
+        -newkey rsa:2048 \
+        -keyout "${KEYFILE}" \
+        -keyform PEM \
+        -nodes \
+        -out "${CERTFILE}" \
+        -outform PEM \
+        -subj "${MACHINE_ID:+/O=${MACHINE_ID}}/CN=${HOSTNAME}" \
+        -config - \
+        -extensions v3_req << EOF
+    [ req ]
+    req_extensions = v3_req
+    extensions = v3_req
+    distinguished_name = req_distinguished_name
+    [ req_distinguished_name ]
+    [ v3_req ]
+    subjectAltName=IP:127.0.0.1,DNS:localhost
+    basicConstraints = critical, CA:TRUE
+    keyUsage = critical, digitalSignature,cRLSign,keyCertSign,keyEncipherment,keyAgreement
+    extendedKeyUsage = serverAuth
+EOF
+}
+
+selfsign() {
+    # Common variables used by both methods
+    local MACHINE_ID="$(tr -c '[:xdigit:][:space:]' '.' < /etc/machine-id)"
+    local HOSTNAME="${HOSTNAME:-$(hostname)}"
+    local CERTFILE="0-self-signed.cert"
+    local KEYFILE="0-self-signed.key"
+    local CA_FILE="0-self-signed-ca.pem"
+
+    # We renew certificates up to 30 days before expiry, so give ourselves a
+    # year, plus 30 days.  The maximum is variously mentioned to be 397 or 398.
+    DAYS=395
```

Comment:
Wouldn't that make more sense a the top of the file, as a global constant? It's a bit weird that other functions access this and it looks like a global var, but isn't.

Reply:
I actually want to mark this `local`.  `local` in shell is a bit weird: it contains the scope of the variable to the defining function, but called functions can still access it, and indeed, just like the names of the files to create, this variable is used in common to both of the self-sign cases, but not the IPA case, so it belongs here.